packages/agent/src/did-manager.ts

```diff
@@ -0,0 +1,371 @@
+    if (did) return did;
+
+    return undefined;
+  }
+
+  async import(options: ImportDidOptions): Promise<ManagedDid> {
+    let { alias, context, did, kms } = options;
+
+    if (did.keySet === undefined) {
+      throw new Error(`Portable DID is missing required property: 'keySet'`);
+    }
+
+    // Verify the DID method is supported.
+    const parsedDid = utils.parseDid({ didUrl: did.did });
+    if (!parsedDid) {
+      throw new Error(`DidManager: Unable to resolve: ${did}`);
+    }
+    const { method } = parsedDid;
+    this.getMethod(method);
+
+    /** Import key set to KeyManager, or if already in KeyManager, retrieve the
+     * public key. */
+    const keySet = await this.importOrGetKeySet({ keySet: did.keySet, kms });
+
+    // Set the KeyManager alias for each key to the DID Document primary ID.
+    await this.updateKeySet({
+      canonicalId : did.canonicalId,
+      didDocument : did.document,
+      keySet
+    });
+
+    // Format the PortableDid and given input as a ManagedDid.
+    const managedDid = { alias, method, ...did, keySet };
+
+    /** If context is undefined, then the DID will be stored under the
+     * tenant of the imported DID. Otherwise, the DID record will
+     * be stored under the tenant of the specified context. */
+    context ??= managedDid.did;
+
+    // Store the ManagedDid in the store.
+    await this._store.importDid({ did: managedDid, agent: this.agent, context });
+
+    return managedDid;
+  }
+
+  /**
+   * Retrieves a `DidMethodApi` instance associated with a specific method
+   * name. This method uses the method name to access the `didMethods` map
+   * and returns the corresponding `DidMethodApi` instance. If a method
+   * name is provided that does not exist within the `didMethods` map, it
+   * will throw an error.
+   *
+   * @param methodName - A string representing the name of the method for
+   * which the corresponding `DidMethodApi` instance is to be retrieved.
+   *
+   * @returns The `DidMethodApi` instance that corresponds to the provided
+   * method name. If no `DidMethodApi` instance corresponds to the provided
+   * method name, an error is thrown.
+   *
+   * @throws Will throw an error if the provided method name does not
+   * correspond to any `DidMethodApi` instance within the `didMethods` map.
+   */
+  private getMethod(methodName: string): DidMethodApi {
+    const didMethod = this._didMethods.get(methodName);
+
+    if (didMethod === undefined) {
+      throw new Error(`The DID method '${methodName}' is not supported`);
+    }
+
+    return didMethod;
+  }
+
+  private async importOrGetKeySet(options: {
+    keySet: DidKeySet,
+    kms: string | undefined
+  }): Promise<DidKeySet> {
+    const { kms } = options;
+
+    // Get the agent instance.
+    const agent = this.agent;
+
+    // Make a deep copy of the key set to prevent side effects.
+    const keySet = structuredClone(options.keySet);
```

Comment:
Wondering if there's an alternative for `structuredClone` we might be able to use that would accomplish the same functionality? 

There's no equivalent for this function in hermes (React Native). I was able to verify that I'm able to polyfill it using a third-party library (https://www.npmjs.com/package/realistic-structured-clone). However, just wanted to see if there was an alternative to avoid any potential compatibility issues.

Reply:
Ideally we keep it since its fully supported in Node.js, browsers, and browser service workers.  There are several third party libs that allow for deep copying of an object if it becomes a show stopper.  I haven't benchmarked to see how much faster `structuredClone` is versus those deep copy libs.